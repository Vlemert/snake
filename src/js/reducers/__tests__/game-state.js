import actionTypes from 'constants/action-types';
import gameStates from 'constants/game-state';
import gameState from '../game-state';

test('Default state is correct', () => {
  expect(gameState()).toMatchSnapshot();
});

test('Handles START_GAME', () => {
  expect(gameState(undefined, {
    type: actionTypes.START_GAME
  })).toBe(gameStates.PLAYING);
});

test('Handles SNAKE_DIED', () => {
  expect(gameState(gameStates.PLAYING, {
    type: actionTypes.SNAKE_DIED
  })).toBe(gameStates.ENDED);
});
