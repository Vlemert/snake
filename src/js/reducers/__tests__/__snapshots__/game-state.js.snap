exports[`test Default state is correct 1`] = `"WELCOME"`;
